protocols/imap/src/main/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParser.java

```diff
@@ -29,119 +28,99 @@
-                if (pos >= DEPTH.length) {
-                    return false;
-                } else {
-                    return DEPTH[pos++] == ImapRequestLineReader.cap(chr);
-                }
-            }
-        });
-        return Depth.fromString(request.atom());
-    }
-    public static void main(String[] agrs) {
-        System.out.println(Depth.fromString("-1"));
+    private Depth consumeDepthOpt(ImapRequestLineReader requestReader) throws DecodingException {
+        if (requestReader.atom().equalsIgnoreCase(DEPTH)) {
+            return Depth.fromString(requestReader.atom());
+        }
+
+        throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, "Wrong on, it should be DEPTH");
     }
 
-    private long consumeMaxsizeOpt(ImapRequestLineReader request) throws DecodingException {
-        request.consumeWord(new CharacterValidator() {
-            int pos = 0;
-            public boolean isValid(char chr) {
-                if (pos >= MAXSIZE.length) {
-                    return false;
-                } else {
-                    return MAXSIZE[pos++] == ImapRequestLineReader.cap(chr);
-                }
-            }
-        });
-        return request.number(true);
+    private long consumeMaxsizeOpt(ImapRequestLineReader requestReader) throws DecodingException {
+        if (requestReader.atom().equalsIgnoreCase(MAXSIZE)) {
+            return requestReader.number(true);
+        }
+
+        throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, "Wrong on, it should be MAXSIZE");
     }
 
-    private void consumeKey(ImapRequestLineReader request, GetAnnotationRequest getAnnotationRequest) throws DecodingException {
-        getAnnotationRequest.setKeys(ImmutableSet.of(request.atom()));
-        request.eol();
+    private void consumeKey(ImapRequestLineReader requestReader, GetAnnotationRequest.GetAnnotationRequestBuilder builder) throws DecodingException {
+        builder.keys(ImmutableSet.of(createAnnotationKey(requestReader.atom())));
+        requestReader.eol();
     }
 
-    private Set<String> consumeKeys(ImapRequestLineReader request) throws DecodingException {
+    private Set<String> consumeKeys(ImapRequestLineReader requestReader) throws DecodingException {
         Builder<String> keys = ImmutableSet.builder();
         do {
-            keys.add(request.atom());
-        } while (request.nextWordChar() != ')');
+            keys.add(createAnnotationKey(requestReader.atom()));
+        } while (requestReader.nextWordChar() != ')');
 
         return keys.build();
     }
 
-    private char nextNonSpaceChar(ImapRequestLineReader request) throws DecodingException {
-        char next = request.nextChar();
-        while (next == ' ') {
-            request.consume();
-            next = request.nextChar();
-        }
-        return next;
+    private MailboxAnnotation.MailboxAnnotationKey createAnnotationKey(String key) throws DecodingException {
```

Comment:
This call do not have much value IMO.
I guess it would be better not to extract it.